### PR TITLE
fix(docker): get container logs even on error

### LINKED ISSUE #211: Docker errors are hidden
In `codegen.py`, `auto_remove` is set to `True` when running the pip install command. This results in the process exiting before stderr can be read, so for example you see something like this:

```
docker.errors.ContainerError: Command 'pip install --no-cache-dir --no-color --disable-pip-version-check --upgrade --requirement /project/requirements.txt --target /project/build' in image 'public.ecr.aws/lambda/python:3.7' returned non-zero exit status 1
```

If you change it to `auto_remove=False`, you get a useful error message:

```
docker.errors.ContainerError: Command 'pip install --no-cache-dir --no-color --disable-pip-version-check --upgrade --requirement /project/requirements.txt --target /project/build' in image 'public.ecr.aws/lambda/python:3.7' returned non-zero exit status 1: b"ERROR: Could not install packages due to an OSError: [Errno 28] No space left on device: '/tmp/pip-unpack-zw24lvnb'\n\n"
```

(The fix for this particular error is to run `docker image prune`)

### PR DESCRIPTION
**Pull request by mmaeng**
Closes #211 

On docker container error `auto_remove=True` would not return container output.  Switched to `remove=True` in `docker.client.containers.run` for slightly less robust container removal but much better logging.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by kddejong — COMMENTED** [review]
A little torn on this change.  Can you put in more details about what this change means?  

If I'm reading correctly if for some reason my process crashes then the container won't be removed anymore?

**Review by ericzbeard — APPROVED** [review]

**Comment by mmaeng** [comment]
That is correct.  If the container exits with an error the container would not be removed so further investigation can be carried out.

**Review by ericzbeard — APPROVED** [review]

**Review by ericzbeard — COMMENTED** [review]
I don't think this will work, based on my testing when I had the issue with space running out on the device.

File: python/rpdk/python/codegen.py [hunk 1]
```diff
@@ -339,6 +347,8 @@ def _docker_build(cls, external_path):
             cause.__cause__ = e
             raise DownstreamError("Error running docker build") from cause
         except (ContainerError, ImageLoadError, APIError) as e:
+            for line in logs:
```
Comment (kddejong) [inline python/rpdk/python/codegen.py]: Similar to 344.  Function?

Comment (mmaeng) [reply python/rpdk/python/codegen.py]: Just for debugging.  Wanted to capture odd usecases in the logs, strange mismatches in pip packages etc.

**Review by kddejong — COMMENTED** [review]
Is this process wrapped in the same process that prevents cancelling it?

**Comment by kddejong** [comment]
I'm still super torn on this PR.  I prefer `auto_remove` and this is a workaround because of an issue with `docker-py` to cover a very specific case which is a container not starting (in this case the disk being full).  I'm just not sure the value is there to do this change.

**Comment by mmaeng** [comment]
Let me clean up this PR

**Pull request closed without merging**